web/server/models.py

```diff
@@ -362,3 +374,145 @@ class ArtifactChange(BaseModel):
     path: str
     type: t.Optional[ArtifactType] = None
     file: t.Optional[File] = None
+
+
+class ReportBackfillProgress(BaseModel):
+    completed: int
+    total: int
+    view_name: str
+    start: TimeLike
+    end: t.Optional[TimeLike] = None
+
+
+class ReportTestsResult(BaseModel):
+    message: str
+
+
+class ReportTestDetails(ReportTestsResult):
+    details: str
+
+
+class ReportTestsFailure(ReportTestsResult):
+    total: int
+    failures: int
+    errors: int
+    successful: int
+    dialect: str
+    details: t.List[ReportTestDetails]
+    traceback: str
+
+
+class BackfillDetails(BaseModel):
+    model_name: t.Optional[str] = None
+    view_name: str
+    interval: t.Tuple[str, str]
+    batches: int
+
+
+class BackfillTask(BaseModel):
+    completed: int
+    total: int
+    view_name: str
+    start: TimeLike
+    end: t.Optional[TimeLike] = None
+
+
+class TrackableMeta(BaseModel):
+    """Keep track of start and end time. Calculate duration and record a status"""
+
+    status: Status = Status.init
+    start: int = Field(default_factory=now_timestamp)
+    end: t.Optional[int] = None
+    done: bool = False
+
+    @property
+    def duration(self) -> int | None:
+        return self.end - self.start if self.start and self.end else None
+
+    def dict(self, *args: t.Any, **kwargs: t.Any) -> t.Dict[str, t.Any]:
+        data = super().dict(*args, **kwargs)
+        data["duration"] = self.duration
+        return data
+
+
+class TrackableMixin(BaseModel):
```

Comment:
If this is a mixin and not meant to be instantiated, it doesn't have to subclass BaseModel. It'd be used like
```python
PlanStageTracker(TrackableMixin, BaseModel):
    pass
```